### PR TITLE
[ranges] Mark exposition-only names as kebab-case.

### LINKED ISSUE #3097: kebab-case all the exposition only ranges types
LWG asked for the exposition-only types in namespace ranges to universally be changed to _kebab-case_ in Cologne, so that it's easier to tell which types are public-facing, and which are exposition-only.

I've volunteered to make that change, but it's a good idea to make an issue to track it.

### PR DESCRIPTION
**Pull request by jensmaurer**
Fixes #3097.

### CODE REVIEW

**Comment by johelegp** [comment]
I'm still preparing a review.

File: source/ranges.tex [hunk 1]
```diff
@@ -6186,117 +6185,117 @@
...
 \effects
-Initializes \tcode{current_} with \tcode{std::move(i.current_)}.
+Initializes \exposid{current_} with \tcode{std::move(i.\exposid{current_})}.
 \end{itemdescr}
 
 \indexlibrarymember{base}{elements_view::iterator}%
 \begin{itemdecl}
-constexpr iterator_t<base_t> base() const &
-  requires copyable<iterator_t<base_t>>;
+constexpr iterator_t<@\exposid{base-t}@> base() const &
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
constexpr iterator_t<@\exposid{base-t}@> base() const&
```

File: source/ranges.tex [hunk 2]
```diff
@@ -6186,117 +6185,117 @@
...
 \pnum
 \effects
-Equivalent to: \tcode{return current_;}
+Equivalent to: \tcode{return \exposid{current_};}
 \end{itemdescr}
 
 \indexlibrarymember{base}{elements_view::iterator}%
 \begin{itemdecl}
-constexpr iterator_t<base_t> base() &&;
+constexpr iterator_t<@\exposid{base-t}@> base() &&;
```
Comment (johelegp) [inline source/ranges.tex]: The return statement below should use `\exposid{current_}`.

File: source/ranges.tex [hunk 3]
```diff
@@ -6307,22 +6306,22 @@
...
-++current_;
+++@\exposid{current_}@;
 return *this;
 \end{codeblock}
 \end{itemdescr}
 
 \indexlibrarymember{operator++}{elements_view::iterator}%
 \begin{itemdecl}
-constexpr void operator++(int) requires (!forward_range<base_t>);
+constexpr void operator++(int) requires (!forward_range<@\exposid{base-t}@>);
```
Comment (johelegp) [inline source/ranges.tex]: Same as above.

File: source/ranges.tex [hunk 4]
```diff
@@ -3071,7 +3071,7 @@
 
 \pnum
 \returns
-\tcode{\{*this, ranges::find_if(base_, ref(*pred_))\}}.
+\tcode{\{*this, ranges::find_if(\exposid{base_}, ref(*pred_))\}}.
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
\tcode{\{*this, ranges::find_if(\exposid{base_}, ref(*\exposid{pred_}))\}}.
```

File: source/ranges.tex [hunk 5]
```diff
@@ -2667,60 +2667,60 @@
...
 \expects
-\tcode{parent_->stream_ != nullptr} is \tcode{true}.
+\tcode{\exposid{parent_}->\exposid{stream_} != nullptr} is \tcode{true}.
 
 \pnum
 \effects
 Equivalent to:
 \begin{codeblock}
-*parent_->stream >> parent_->object_;
+*@\exposid{parent_}@->stream >> @\exposid{parent_}@->@\exposid{object_}@;
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
*@\exposid{parent_}@->@\exposid{stream_}@ >> @\exposid{parent_}@->@\exposid{object_}@;
```

File: source/ranges.tex [hunk 6]
```diff
@@ -1603,18 +1603,18 @@
...
+  @\exposid{size_}@ -= @\placeholdernc{make-unsigned-like}@(d);
 else
-  size_ += @\placeholdernc{make-unsigned-like}@(-d);
+  @\exposid{size_}@ += @\placeholdernc{make-unsigned-like}@(-d);
 return *this;
 \end{codeblock}
 \item Otherwise,
 \begin{codeblock}
-ranges::advance(begin_, n, end_);
+ranges::advance(begin_, n, @\exposid{end_}@);
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
ranges::advance(@\exposid{begin_}@, n, @\exposid{end_}@);
```

File: source/ranges.tex [hunk 7]
```diff
@@ -1603,18 +1603,18 @@
 \effects
 Equivalent to:
 \begin{itemize}
-\item If \tcode{StoreSize} is \tcode{true},
+\item If \exposid{StoreSize} is \tcode{true},
 \begin{codeblock}
-auto d = n - ranges::advance(begin_, n, end_);
+auto d = n - ranges::advance(begin_, n, @\exposid{end_}@);
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
auto d = n - ranges::advance(\exposid{begin_}, n, @\exposid{end_}@);
```

File: source/ranges.tex [hunk 8]
```diff
@@ -4224,68 +4224,68 @@
...
+    @\exposid{sentinel}@() = default;
+    constexpr explicit @\exposid{sentinel}@(sentinel_t<@\exposid{Base}@> end);
+    constexpr @\exposid{sentinel}@(@\exposid{sentinel}@<!Const> s)
+      requires Const && convertible_to<sentinel_t<V>, sentinel_t<@\exposid{Base}@>>;
 
-    constexpr sentinel_t<Base> base() const;
+    constexpr sentinel_t<@\exposid{Base}@> base() const;
 
-    friend constexpr bool operator==(const CI& y, const sentinel& x);
+    friend constexpr bool operator==(const CI& y, const @\exposid{sentinel}@& x);
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
    friend constexpr bool operator==(const @\exposid{CI}@& y, const @\exposid{sentinel}@& x);
```

File: source/ranges.tex [hunk 9]
```diff
@@ -4224,68 +4224,68 @@
...
 \pnum
 \effects
-Equivalent to: \tcode{return end_;}
+Equivalent to: \tcode{return \exposid{end_};}
 \end{itemdescr}
 
 \indexlibrarymember{operator==}{take_view::sentinel}%
 \begin{itemdecl}
-friend constexpr bool operator==(const CI& y, const sentinel& x);
+friend constexpr bool operator==(const CI& y, const @\exposid{sentinel}@& x);
```
Comment (johelegp) [inline source/ranges.tex]:
```suggestion
friend constexpr bool operator==(const @\exposid{CI}@& y, const @\exposid{sentinel}@& x);
```

File: source/ranges.tex [hunk 10]
```diff
@@ -3108,21 +3108,21 @@
...
-    friend constexpr range_rvalue_reference_t<V> iter_move(const iterator& i)
-      noexcept(noexcept(ranges::iter_move(i.current_)));
-    friend constexpr void iter_swap(const iterator& x, const iterator& y)
-      noexcept(noexcept(ranges::iter_swap(x.current_, y.current_)))
-      requires indirectly_swappable<iterator_t<V>>;
+    friend constexpr range_rvalue_reference_t<V> iter_move(const @\exposid{iterator}@& i)
+      noexcept(noexcept(ranges::iter_move(i.@\exposid{current_}@)));
+    friend constexpr void iter_swap(const @\exposid{iterator}@& x, const @\exposid{iterator}@& y)
+      noexcept(noexcept(ranges::iter_swap(x.@\exposid{current_}@, y.@\exposid{current_}@)))
+      requires indirectly_swappable<@\exposid{iterator}@_t<V>>;
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex [hunk 11]
```diff
@@ -3338,29 +3338,29 @@
...
 \begin{itemdecl}
-constexpr explicit sentinel(filter_view& parent);
+constexpr explicit @\exposid{sentinel}@(filter_view& parent);
 \end{itemdecl}
 
 \begin{itemdescr}
 \pnum
 \effects
-Initializes \tcode{end_} with \tcode{ranges::end(parent.base_)}.
+Initializes \exposid{end_} with \tcode{ranges::end(\exposid{parent_}.\exposid{base_})}.
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex [hunk 12]
```diff
@@ -3591,85 +3591,85 @@
...
+      remove_cvref_t<invoke_result_t<F&, range_reference_t<@\exposid{Base}@>>>;
+    using difference_type   = range_difference_t<@\exposid{Base}@>;
 
-    iterator() = default;
-    constexpr iterator(Parent& parent, iterator_t<Base> current);
-    constexpr iterator(iterator<!Const> i)
-      requires Const && convertible_to<iterator_t<V>, iterator_t<Base>>;
+    @\exposid{iterator}@() = default;
+    constexpr @\exposid{iterator}@(@\exposid{Parent}@& parent, iterator_t<@\exposid{Base}@> current);
+    constexpr @\exposid{iterator}@(iterator<!Const> i)
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex [hunk 13]
```diff
@@ -4224,68 +4224,68 @@
...
   template<class V>
   template<bool Const>
-  class take_view<V>::sentinel {
+  class take_view<V>::@\exposid{sentinel}@ {
   private:
-    using Base = conditional_t<Const, const V, V>;      // \expos
-    using CI = counted_iterator<iterator_t<Base>>;      // \expos
-    sentinel_t<Base> end_ = sentinel_t<Base>();         // \expos
+    using @\exposid{Base}@ = conditional_t<Const, const V, V>;      // \expos
+    using CI = counted_iterator<iterator_t<@\exposid{Base}@>>;      // \expos
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex [hunk 14]
```diff
@@ -4333,31 +4333,31 @@
   requires input_range<V> && is_object_v<Pred> &&
     indirect_unary_predicate<const Pred, iterator_t<V>>
   class take_while_view : public view_interface<take_while_view<V, Pred>> {
-    template<bool> class sentinel;                      // \expos
+    template<bool> class @\exposid{sentinel}@;                      // \expos
 
-    V base_;                                            // \expos
-    @\placeholder{semiregular-box}@<Pred> pred_; @\itcorr[-1]@                       // \expos
+    V @\exposid{base_}@;                                            // \expos
+    @\placeholder{semiregular-box}@<Pred> @\exposid{pred_}@; @\itcorr[-1]@                       // \expos
```
[the remark this answers is not in the document]
Comment (jensmaurer) [reply source/ranges.tex]: This is trying to align the comment that follows with the other comments in the surroundings. This is still substantially broken, but likely benefits from a more holistic approach. Not now.

File: source/ranges.tex (hunk 5 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 6 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 7 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 8 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

File: source/ranges.tex (hunk 9 above)
Comment (jensmaurer) [reply source/ranges.tex]: Fixed.

**Comment by jensmaurer** [comment]
Fixes are in #3562.

**Pull request merged**